pkg/scaling/resolver/hashicorpvault_handler.go

```diff
@@ -28,6 +29,16 @@ func NewHashicorpVaultHandler(v *kedav1alpha1.HashiCorpVault) *HashicorpVaultHan
 // Initialize the Vault client
 func (vh *HashicorpVaultHandler) Initialize(logger logr.Logger) error {
 	config := vaultApi.DefaultConfig()
+
+	// do not take into account vault namespace env variable
+	// It prevents any conflict with env during testing, just use HashiCorpVault inputs.
+	// Vault client creation use env var to override any conf (NewClient)
+	// We should probably also unset EnvVaultToken for the same purpose.
+	err := os.Unsetenv(vaultApi.EnvVaultNamespace)
```

Comment:
Is this something that would be set automatically on all pods including KEDA, or is it something the user is explicitly setting?

Reply:
Vault handles several way to provide token and namespace with a precedence (explicit overriden by file and overriden by env var).
I'm working also on Kubernetes appliance and we've integrated Vault Enterprise. We've noticed that when using explicit namespace and/or token configuration (from a Kube spec in our example), it's really painful for developers when testing. They may have configured some Vault namespace locally using an env var and it overrides the conf in the tested soft.
It's probably a bad idea I had to also push that code in that PR, I propose to remove that code and eventually, If you want to, provide another PR to ignore both VAULT_NAMESPACE and VAULT_TOKEN env variable.